lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -238,145 +274,248 @@ bool Caffe2ModelLoader::hasMultidirectionalBroadcast(
-      int32_t offset;
-      ASSIGN_VALUE_OR_RETURN_ERR(offset, loadInt(dict["Y_zero_point"]));
-      outTy = G_.getParent()->uniqueType(ElemKind::Int8QTy, outDims, scale,
-                                         offset - OFFSETSHIFT);
-    }
+  TypeRef finalInType = finalIn.getType();
 
-    Node *node = G_.createConv(opName, finalIn, filter, bias, outTy, kernels,
-                               strides, pads, group);
-    if (typeName == "ConvRelu") {
-      node = G_.createRELU(opName + ".relu", node);
-    }
-    if (order == "NCHW") {
-      // Transpose the output back.
-      node = G_.createTranspose(opName, node, NHWC2NCHW);
+  // Calculate the size and allocate the output buffer.
+  ShapeNHWC idim = ShapeNHWC(finalInType->dims());
+  auto outSz =
+      calculateConvPoolOutputDims(idim.h, idim.w, kernels, strides, pads);
+  std::array<size_t, 4> outDims = {{idim.n, outSz.first, outSz.second, depth}};
+
+  TypeRef outTy;
+
+  RETURN_ERR_IF_NOT(dict.count("Y_zero_point"),
+                    "missing zero point for quantized output type");
+  RETURN_ERR_IF_NOT(dict.count("Y_scale"),
+                    "missing Y_scale for quantized output type");
+
+  // Try to find a loaded bias constant.
+  Constant *bias = nullptr;
+  if (op.input_size() > 2) {
+    const auto &biasName = op.input(2);
+    bias = getConstantByNameOrNull(biasName);
+  }
+  // Construct the bias constant if one wasn't found.
+  if (!bias) {
+    Tensor b(ElemKind::Int32QTy, {depth}, 1.0, 0);
+    b.zero();
+    bias = G_.getParent()->createConstant("conv.bias", std::move(b));
+  }
+
+  RETURN_ERR_IF_NOT(bias->getPayload().size() == depth,
+                    "Loaded bias tensor of incorrect size");
+
+  // Construct output type
+  float scale;
+  ASSIGN_VALUE_OR_RETURN_ERR(scale, loadFloat(dict["Y_scale"]));
+  int32_t offset;
+  ASSIGN_VALUE_OR_RETURN_ERR(offset, loadInt(dict["Y_zero_point"]));
+  outTy = G_.getParent()->uniqueType(ElemKind::Int8QTy, outDims, scale,
+                                     offset - OFFSETSHIFT);
+
+  Node *node;
+
+  if (quantizeGroupwise) {
+    auto wScalesName = strFormat("%s_loaded_scales", op.input(1).c_str());
+    auto wOffsetsName = strFormat("%s_loaded_offsets", op.input(1).c_str());
+    Constant *wScales;
+    Constant *wOffsets;
+    ASSIGN_VALUE_OR_RETURN_ERR(wScales, getConstantByName(wScalesName));
+    ASSIGN_VALUE_OR_RETURN_ERR(wOffsets, getConstantByName(wOffsetsName));
+
+    node = G_.createChannelwiseQuantizedConv(opName, finalIn, w, bias, wScales,
+                                             wOffsets, outTy, kernels, strides,
+                                             pads, group);
+  } else {
+    // If the bias isn't quantized for a non group quantized conv, quantize it.
+    const Tensor &biasTensor = bias->getPayload();
+    if (biasTensor.getElementType() == ElemKind::FloatTy) {
+      TensorQuantizationParams tqp;
+      tqp.offset = 0;
+      // input scale * filter scale
```

Comment:
Can you expand a little here? As is the comment doesn't bring much value, since it's clear this is what's happening on the next line.